### PR TITLE
Bs new

### PR DESCRIPTION
**Pull request by dinojugosloven**
New beam selector stuff.

Note: running pytest tests failed in bs-new.

### CODE REVIEW

File: malcolm/modules/scanning/controllers/runnablecontroller.py [hunk 1]
```diff
@@ -10,9 +10,9 @@
 from ..infos import ParameterTweakInfo, RunProgressInfo, ConfigureParamsInfo
 from ..util import RunnableStates, AGenerator, ConfigureParams
 from ..hooks import ConfigureHook, ValidateHook, PostConfigureHook, \
-    PreRunHook, RunHook, PostRunArmedHook, PostRunReadyHook, \
-    ReportStatusHook, AbortHook, PauseHook, SeekHook, ControllerHook, \
-    PreConfigureHook, AAxesToMove
+    RunHook, PostRunArmedHook, PostRunReadyHook, ResumeHook, ReportStatusHook, \
```
Comment (thomascobb) [inline malcolm/modules/scanning/controllers/runnablecontroller.py]: Please restore this file

File: malcolm/modules/scanning/hooks.py [hunk 2]
```diff
@@ -232,5 +228,9 @@ def __init__(self,
             axesToMove=axesToMove, **kwargs)
 
 
+class ResumeHook(ControllerHook[None]):
```
Comment (thomascobb) [inline malcolm/modules/scanning/hooks.py]: Please restore this file

File: malcolm/modules/scanning/blocks/shutter_block.yaml [hunk 3]
```diff
@@ -1,23 +0,0 @@
-- builtin.parameters.string:
```
Comment (thomascobb) [inline malcolm/modules/scanning/blocks/shutter_block.yaml]: Please restore this file

File: malcolm/modules/scanning/parts/minturnaroundpart.py [hunk 4]
```diff
@@ -7,35 +7,26 @@
...
 with Anno("Initial value for min time between non-joined points"):
     AMinTurnaround = float
-with Anno("Minimum interval between turnaround points"):
-    ATurnaroundInterval = float
 
 
 class MinTurnaroundPart(Part):
-    def __init__(self, name="minTurnaround", gap=None, interval=None):
-        # type: (APartName, AMinTurnaround, ATurnaroundInterval) -> None
+    def __init__(self, name="minTurnaround", value=None):
```
Comment (thomascobb) [inline malcolm/modules/scanning/parts/minturnaroundpart.py]: Please restore this file

File: tests/test_modules/test_pmac/test_motorinfo.py [hunk 5]
```diff
@@ -54,7 +54,7 @@ def test_settle(self):
         distance = 0.1
         self.o.velocity_settle = 0.1
         p = self.o.make_velocity_profile(
-            v1, v2, distance, 0.0, 0.001)
+            v1, v2, distance, 0.0)
```
Comment (thomascobb) [inline tests/test_modules/test_pmac/test_motorinfo.py]: Please restore this file

File: tests/test_modules/test_pmac/test_velocityprofile.py [hunk 6]
```diff
@@ -32,7 +32,7 @@ def do_test_time_range(v1, v2, quantize=False):
             ts = np.arange(.01, 20, .1)
         for t in ts:
             d = 100
-            profile = VelocityProfile(v1, v2, d, t, 2.0, 10000, interval=.002)
+            profile = VelocityProfile(v1, v2, d, t, 2.0, 10000)
```
Comment (thomascobb) [inline tests/test_modules/test_pmac/test_velocityprofile.py]: Please restore this file

File: tests/test_modules/test_scanning/test_shutter_block.py [hunk 7]
```diff
@@ -1,12 +0,0 @@
-from mock import Mock
```
Comment (thomascobb) [inline tests/test_modules/test_scanning/test_shutter_block.py]: Please restore this file

File: tests/test_modules/test_scanning/test_unrollingpart.py [hunk 8]
```diff
@@ -66,29 +66,32 @@ def test_2d_no_changes_needed_1dim(self):
         results = self.b.validate(generator, ["x"])
         assert generator == results["generator"]
 
+    @pytest.mark.skip(reason="need 'alternate' scanpointgenerator changes")
     def test_3d_generator(self):
-        generator = make_generator(include_z=True)
+        generator = make_generator()
```
Comment (thomascobb) [inline tests/test_modules/test_scanning/test_unrollingpart.py]: Please restore this file

**Comment by thomascobb** [comment]
Duplicate of #283

**Comment by dinojugosloven** [comment]
I am going to delete this unmerged branch (bs-new) on the server, if you
are OK with it.

Cheers,
Dino

**Comment by thomascobb** [comment]
Yes that's fine

Sent from my phone

________________________________

**Pull request closed without merging**